src/app/components/experiment-timeline/experiment-timeline.component.html

```diff
@@ -1,12 +1,60 @@
 <div class="big-content">
     <div class="flex-header">
-        <h1 class="t-page-headline">Experiment Timeline</h1>
+        <h1 class="t-page-headline">Experiment Timeline
+            <mat-icon class="arrow" (click)="onSort()">
+                {{sort>=0?"arrow_upward":"arrow_downward"}}
+            </mat-icon>
+        </h1>
 
-        <mat-paginator class="paginator" (page)="onPageChange($event)" [length]="collectionSize" [pageSize]="10"
-            [pageSizeOptions]="[5, 10, 25, 100]" aria-label="Select page">
+        <mat-paginator #paginatorTop class="paginator"
+            (page)="paginatorBottom.pageSize = $event.pageSize; paginatorBottom.pageIndex = $event.pageIndex; onPageChange($event)"
+            [length]="collectionSize" [pageSize]="10" [pageSizeOptions]="[5, 10, 25, 100]" aria-label="Select page">
         </mat-paginator>
     </div>
 
+
+    <div class="filter">
+        <mat-expansion-panel [style.width.%]="100">
+            <mat-expansion-panel-header>
+                <mat-panel-title>
+                    Filter Timeline Steps
+                </mat-panel-title>
+            </mat-expansion-panel-header>
+            <mat-form-field [style.width.%]="100">
+                <mat-label>Search plugin name</mat-label>
+                <input [(ngModel)]="pluginName" matInput (keydown.enter)="updatePageContent()">
+                <button mat-button matSuffix class="search-button" (click)="updatePageContent()">Ok</button>
+            </mat-form-field>
+
+            <mat-form-field [style.width.%]="100">
+                <mat-label>Search version</mat-label>
+                <input [(ngModel)]="version" matInput (keydown.enter)="updatePageContent()">
+                <button mat-button matSuffix class="search-button" (click)="updatePageContent()">Ok</button>
+            </mat-form-field>
+
+            <mat-form-field [style.width.%]="100">
+                <mat-label>Status</mat-label>
+                <mat-select [(ngModel)]="stepStatus" name="status" (selectionChange)="updatePageContent()">
+                    <mat-option *ngFor="let status of statusValues" [value]="status">
+                        {{status}}
+                    </mat-option>
+                </mat-select>
+            </mat-form-field>
+
+            <mat-form-field [style.width.%]="100">
+                <mat-label>Cleared/uncleared substeps</mat-label>
+                <mat-select [(ngModel)]="unclearedSubstep" name="unclearedSubstep"
+                    (selectionChange)="updatePageContent()">
+                    <mat-option *ngFor="let unclearedSubstepValue of unclearedSubstepValues"
```

Comment:
Since there are only three options a radio button group or a button cycling through the options would also be an option to consider here.